docs/fsharp/language-reference/type-extensions.md

```diff
@@ -1,99 +1,151 @@
 ---
 title: Type Extensions (F#)
 description: Learn how F# type extensions allow you add new members to a previously defined object type.
-ms.date: 05/16/2016
+ms.date: 07/14/2018
 ---
 # Type Extensions
 
-Type extensions let you add new members to a previously defined object type.
+Type extensions (also called _augmentations_) are a family of features that let you add new members to a previously defined object type. The three features are:
+
+* Intrinsic type extensions
+* Optional type extensions
+* Extension methods
+
+Each can be used in different scenarios and has different tradeoffs.
 
 ## Syntax
 
 ```fsharp
-// Intrinsic extension.
+// Intrinsic and optional extensions
 type typename with
     member self-identifier.member-name =
         body
     ...
-[ end ]
 
-// Optional extension.
-type typename with
-    member self-identifier.member-name =
+// Extension methods
+open System.Runtime.CompilerServices
+
+// On a module
+[<Extension>]
+module TypeNameExtensions =
+    let [inline] extension-name (ty: typename) [args] =
+        body
+    ...
+
+// On a class
+[<Extension>]
+type Extensions() =
+    [static] member self-identifier.extension-name (ty: typename, [args]) =
         body
     ...
-[ end ]
 ```
 
-## Remarks
-There are two forms of type extensions that have slightly different syntax and behavior. An *intrinsic extension* is an extension that appears in the same namespace or module, in the same source file, and in the same assembly (DLL or executable file) as the type being extended. An *optional extension* is an extension that appears outside the original module, namespace, or assembly of the type being extended. Intrinsic extensions appear on the type when the type is examined by reflection, but optional extensions do not. Optional extensions must be in modules, and they are only in scope when the module that contains the extension is open.
+## Intrinsic type extensions
 
-In the previous syntax, *typename* represents the type that is being extended. Any type that can be accessed can be extended, but the type name must be an actual type name, not a type abbreviation. You can define multiple members in one type extension. The *self-identifier* represents the instance of the object being invoked, just as in ordinary members.
+An intrinsic type extension is an extension that appears in the same namespace or module, in the same source file, and in the same assembly (DLL or executable file) as the type being extended.
```

Comment:
Same source file implies same Assembly no? Less enumeration helps get the fact it's not just namespace or module but also same source file, so no parity with C# partial classes there. That's really important to underline.

Might even rewrite to something like "An intrinsic extension will only be considered intrinsic if it appears in the same source file and same scope, be that a namespace or module."

Reply:
I think the `and` is supposed to be `or` - this is an old document, so I unfortunately don't know the history.